### ISSUE TITLE
crash in renepay plugin

### ISSUE THREAD

**Issue by endothermicdev** [ElementsProject/lightning#6493]
### Issue and Steps to Reproduce
ran `lightning-cli renepay <bolt 11> 50000 40000` to test a 50sat payment
which lead to the following crash:
```
2023-08-03T14:34:18.711Z UNUSUAL plugin-cln-renepay: onion error WIRE_INCORRECT_CLTV_EXPIRY from node #1 791381x1127x0: failed: WIRE_INCORRECT_CLTV_EXPIRY (reply from remote)
2023-08-03T14:34:18.732Z **BROKEN** plugin-cln-renepay: Underflow subtracting flow->amounts[0] (367309429976msat) from &p->total_sent (0msat)
2023-08-03T14:34:18.763Z INFO    plugin-cln-renepay: Killing plugin: exited during normal operation
2023-08-03T14:34:18.763Z **BROKEN** plugin-cln-renepay: Plugin marked as important, shutting down lightningd!
```
Running 23.08rc1

**Comment by vincenzopalazzo** [comment]
We should not make renepay marked as important, I do not think it has enough tests to do it?

**Comment by Lagrang3** [comment]
> 0

First thing I notice is that invoice string comes before the payment parameters otherwise the parse will not let you continue. 

Can you upload more details of the logs, to simplify maybe just the lines containing `renepay`?

**Comment by Lagrang3** [comment]
Actually, I think I don't need any more log messages. I seems like the `flow` destructor is called twice. I'm looking into it.

**Comment by Lagrang3** [comment]
> We should not make renepay marked as important, I do not think it has enough tests to do it?
> 
> Maybe this underflow is easier for @Lagrang3 to debug? assigning to it till I will be away from my desk

As a builtin plugin, the `bool important` flag is true by default. I am not sure if I can flip it at will. There are conflicting definitions of `struct plugin` in `lightningd/plugin.h` and `plugins/libplugin.c`, the first contains a `bool important`, the second does not.

**Comment by vincenzopalazzo** [comment]
If there is no possibility to do it, we should add it. Till now, all the plugins built-in are important (except grpc one), but we should have the possibility to say, "it is fine if this plugin dies"

**Comment by Lagrang3** [comment]
@rustyrussell just PR this #6523 to avoid crashing the node.

**Comment by Lagrang3** [comment]
@rustyrussell there is also the fix to the underflow that triggered this issued #6505

**Comment by vincenzopalazzo** [comment]
My fault I was assuming that this was already merged

**Comment by rustyrussell** [comment]
I believe this should be fixed by #6538
